spinnaker/spinnaker_system/google_http_lb_test.py

```diff
@@ -0,0 +1,1256 @@
+         jp.PathContainsPredicate('hostRules/hosts', 'host2.com'),
+         jp.PathContainsPredicate('pathMatchers/pathRules/paths', '/golang'),
+         jp.PathContainsPredicate('hostRules/hosts', 'golang.com'),
+       ]
+     )
+    )
+    (contract_builder.new_clause_builder('Forwarding Rule Created',
+                                         retryable_for_secs=30)
+     .list_resource('globalForwardingRules')
+     .contains_pred_list(
+       [
+         jp.PathContainsPredicate('name', self.__lb_name),
+         jp.PathContainsPredicate('portRange', '443-443')
+       ]
+     )
+    )
+    target_proxy_name = '%s-%s' % (self.__lb_name, HTTPS_PROXY_PREFIX)
+    (contract_builder.new_clause_builder('Target Proxy Created',
+                                         retryable_for_secs=30)
+     .list_resource('targetHttpsProxies')
+     .contains_pred_list(
+       [
+         jp.PathContainsPredicate('name', target_proxy_name)
+       ]
+     )
+    )
+
+    return st.OperationContract(
+      self.new_post_operation(title='change cert',
+                              data=payload, path='tasks'),
+      contract=contract_builder.build()
+    )
+
+
+class GoogleHttpLoadBalancerTest(st.AgentTestCase):
+  '''Test fixture for Http LB test.
+  '''
+
+
+  @staticmethod
+  def setUpClass():
+    runner = citest.base.TestRunner.global_runner()
+    bindings = runner.bindings
+    scenario = runner.get_shared_data(GoogleHttpLoadBalancerTestScenario)
+    managed_region = runner.bindings['TEST_GCE_REGION']
+    title = 'Check Quota for {0}'.format(scenario.__class__.__name__)
+
+    verify_results = gcp.verify_quota(
+      title,
+      scenario.gcp_observer,
+      project_quota=GoogleHttpLoadBalancerTestScenario.MINIMUM_PROJECT_QUOTA,
+      regions=[(managed_region,
+                GoogleHttpLoadBalancerTestScenario.MINIMUM_REGION_QUOTA)]
+    )
+    if not verify_results:
+      raise RuntimeError('Insufficient Quota: {0}'.format(verify_results))
+
+    key = crypto.PKey()
+    key.generate_key(crypto.TYPE_RSA, 2048)
+    cert = crypto.X509()
+    cert.get_subject().C = "US"
+    cert.get_subject().ST = "New York"
+    cert.get_subject().L = "New York City"
+    cert.get_subject().O = "Google"
+    cert.get_subject().OU = "Spinnaker"
+    cert.get_subject().CN = "localhost"
+    cert.set_serial_number(4096)
+    cert.gmtime_adj_notBefore(0)
+    cert.gmtime_adj_notAfter(24 * 60 * 60)
+    cert.set_issuer(cert.get_subject())
+    cert.set_pubkey(key)
+    cert.sign(key, 'sha1')
+
+    compute_agent = gcp.GcpComputeAgent.make_agent(scopes=SCOPES,
+                                                   credentials_path=bindings['GCE_CREDENTIALS_PATH'])
+    ssl_cert = {
+      'name': FIRST_CERT,
+      'privateKey': crypto.dump_privatekey(crypto.FILETYPE_PEM, key),
+      'certificate': crypto.dump_certificate(crypto.FILETYPE_PEM, cert)
+    }
+    compute_agent.invoke_resource(ExecutionContext(), # No predicates against this agent, context is empty.
```

Comment:
Use the same execution context for all of the setup.
It doenst matter right now, but if the context were to be used, you'd probably want the same one because the context (setUpClass) is the same.